### ISSUE TITLE
CopyFileFromDevice

### ISSUE THREAD

**Issue by swy680614** [0xFireball/MK.MobileDevice#12]
I try to copy a sharing file from app Documents to pc.
Here is my code
bool ifConnectToVF = iph.ConnectViaHouseArrest("xxx");
if (ifConnectToVF)
{
      String file = @"/Documents/test.txt";
       if (iph.FileExists(file)) {
           bool isSucess = iph.CopyFileFromDevice(@"C:\Users\Public", file);
       }
}                
----------------------------------------------------------------------------------------------
        internal static AFCError copyToDisk(IntPtr afcClient, string filePath, string savePath)
        {
            IntPtr infoPtr;
            AFCError returnCode = afc_get_file_info(afcClient, filePath, out infoPtr);
            if (returnCode != AFCError.AFC_E_SUCCESS)
            {
                return returnCode;  //always AFC_E_OBJECT_NOT_FOUND
            }

How can I fix this?
Thanks.

**Comment by 0xFireball** [comment]
HouseArrest has been having issues since iOS 8.3:
Please see #8 for more information.

Additionally, you don't have to use a custom `copyToDisk` function, the `iPhone` class has that functionality built in. When you connect via House arrest, it should automatically switch to the House Arrest AFC client.

**Comment by swy680614** [comment]
I don't call copyToDisk directly.
I copy file via CopyFileFromDevice and the method "CopyFileFromDevice" will call copyToDisk.
So there is no any solution for copy file from device ?
By the way, this project is really great.

**Comment by 0xFireball** [comment]
@swy680614 you actually can copy files from the device, just not the HouseArrest directories at the moment. HouseArrest SHOULD work if your app specifically allows such access, however.

**Comment by swy680614** [comment]
@0xFireball I can copy files from the device without HouseArrest. But I don't know how. Any sample code I can reference?

**Comment by Glabenwitt** [comment]
I sucessfully used CopyFileFromDevice method but i have to change it's content like that :
`public bool CopyFileFromDevice(string computerFile, string deviceFile)`
`{`
`            iDevice id = Devices[0];`
`            IntPtr currDevice;`
`            string currUdid = id.Udid;`
`            LibiMobileDevice.iDeviceError returnCode = LibiMobileDevice.NewDevice(out currDevice, currUdid);`
`            return AFC.copyToDisk(this.hAFC, deviceFile, computerFile) == AFC.AFCError.AFC_E_SUCCESS;`
`        }`

I hope it can help you

**Comment by 0xFireball** [comment]
Yes,  simply do the same code without calling `ConnectViaHouseArrest`
